driver.go

```diff
@@ -604,6 +605,83 @@ type ocRows struct {
 	options TraceOptions
 }
 
+// HasNextResultSet calls the implements the driver.RowsNextResultSet for ocRows.
+// It returns the the underlying result of HasNextResultSet from the ocRows.parent
+// if the parent implements driver.RowsNextResultSet.
+func (r ocRows) HasNextResultSet() bool {
+	if v, ok := r.parent.(driver.RowsNextResultSet); ok {
+		return v.HasNextResultSet()
+	}
+
+	return false
+}
+
+// NextResultsSet calls the implements the driver.RowsNextResultSet for ocRows.
+// It returns the the underlying result of NextResultSet from the ocRows.parent
+// if the parent implements driver.RowsNextResultSet.
+func (r ocRows) NextResultSet() error {
+	if v, ok := r.parent.(driver.RowsNextResultSet); ok {
+		return v.NextResultSet()
+	}
+
+	return nil
+}
+
+// ColumnTypePrecisionScale calls the implements the driver.RowsColumnTypePrecisionScale for ocRows.
+// It returns the the underlying result of ColumnTypePrecisionScale from the ocRows.parent
+// if the parent implements driver.RowsColumnTypePrecisionScale.
+func (r ocRows) ColumnTypePrecisionScale(index int) (precision, scale int64, ok bool) {
+	if v, ok := r.parent.(driver.RowsColumnTypePrecisionScale); ok {
+		return v.ColumnTypePrecisionScale(index)
+	}
+
+	return 0, 0, false
+}
+
+// ColumnTypeNullable calls the implements the driver.RowsColumnTypeNullable for ocRows.
+// It returns the the underlying result of ColumnTypeNullable from the ocRows.parent
+// if the parent implements driver.RowsColumnTypeNullable.
+func (r ocRows) ColumnTypeNullable(index int) (nullable, ok bool) {
+	if v, ok := r.parent.(driver.RowsColumnTypeNullable); ok {
+		return v.ColumnTypeNullable(index)
+	}
+
+	return false, false
+}
+
+// ColumnTypeLength calls the implements the driver.RowsColumnTypeLength for ocRows.
+// It returns the the underlying result of ColumnTypeLength from the ocRows.parent
+// if the parent implements driver.RowsColumnTypeLength.
+func (r ocRows) ColumnTypeLength(index int) (length int64, ok bool) {
+	if v, ok := r.parent.(driver.RowsColumnTypeLength); ok {
+		return v.ColumnTypeLength(index)
+	}
+
+	return 0, false
+}
+
+// ColumnTypeScanType calls the implements the driver.RowsColumnTypeScanType for ocRows.
+// It returns the the underlying result of ColumnTypeScanType from the ocRows.parent
+// if the parent implements driver.RowsColumnTypeScanType.
+func (r ocRows) ColumnTypeScanType(index int) reflect.Type {
+	if v, ok := r.parent.(driver.RowsColumnTypeScanType); ok {
+		return v.ColumnTypeScanType(index)
+	}
+
+	return nil
```

Comment:
This method implementation is not transparent. The code actually using this interface does the following:

```go
if prop, ok := rowsi.(driver.RowsColumnTypeScanType); ok {
	ci.scanType = prop.ColumnTypeScanType(i)
} else {
	ci.scanType = reflect.TypeOf(new(interface{})).Elem()
}
```

So either return `reflect.TypeOf(new(interface{})).Elem()` (brittle as the go sql code using this functionality might change)  or hide this interface extension when wrapped drivers do not implement this interface